python/pyspark/sql/readwriter.py

```diff
@@ -964,6 +964,21 @@ def _test():
     except py4j.protocol.Py4JError:
         spark = SparkSession(sc)
 
+    hive_enabled = True
+    try:
+        sc._jvm.org.apache.hadoop.hive.conf.HiveConf()
+    except py4j.protocol.Py4JError:
+        hive_enabled = False
+    except TypeError:
+        hive_enabled = False
+
+    if not hive_enabled:
+        # if hive is not enabled, then skip doctests that need hive
+        # TODO: Need to communicate with outside world that this test
+        # has been skipped.
+        m = pyspark.sql.readwriter
+        m.__dict__["DataFrameReader"].__dict__["table"].__doc__ = ""
+
```

Comment:
Thank you for pointer, @HyukjinKwon .

For `readwriter.py`, we had better test without Hive. How do you think , @HyukjinKwon and @bersprockets .
```
- spark = SparkSession.builder.enableHiveSupport().getOrCreate()
+ spark = SparkSession.builder.getOrCreate()
```

Reply:
@bersprockets I was thinking like that but wanted to ask your thought per this PR. I am okay with either way.

Reply:
@HyukjinKwon @dongjoon-hyun I agree, it should go in PR #21141.